packages/dd-trace/src/llmobs/telemetry.js

```diff
@@ -1,12 +1,59 @@
 'use strict'
 
+const {
+  SPAN_KIND,
+  MODEL_PROVIDER,
+  PARENT_ID_KEY,
+  SESSION_ID,
+  ROOT_PARENT_ID,
+  INTEGRATION,
+  DECORATOR
+} = require('./constants/tags')
+
+const ERROR_TYPE = require('../constants')
+
 const telemetryMetrics = require('../telemetry/metrics')
+
+const LLMObsTagger = require('./tagger')
+
 const llmobsMetrics = telemetryMetrics.manager.namespace('mlobs')
 
 function incrementLLMObsSpanStartCount (tags, value = 1) {
   llmobsMetrics.count('span.start', tags).inc(value)
 }
 
+function incrementLLMObsSpanFinishedCount (span, value = 1) {
+  const mlObsTags = LLMObsTagger.tagMap.get(span)
+  const spanTags = span.context()._tags
+
+  const isRootSpan = mlObsTags[PARENT_ID_KEY] === ROOT_PARENT_ID
+  const hasSessionId = mlObsTags[SESSION_ID] != null
+  const integration = mlObsTags[INTEGRATION]
+  const autoInstrumented = integration != null
+  const decorator = !mlObsTags[DECORATOR]
```

Comment:
this is totally my bad but I think it should be this. or we can do `=== true`. apologies! i think i was misreading a bit before between looking at this and python. (`!!` coalesces it into a boolean)
```suggestion
  const decorator = !!mlObsTags[DECORATOR]
```